### ISSUE TITLE
[BUG] MINTEMP on GTR v1.0 with MAX31865

### ISSUE THREAD

**Issue by truglodite** [MarlinFirmware/Marlin#24513]
### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

I'm have been using HW SPI2 with an external max31865 and LCD SD_CARD for a while with marlin 2.0.9.1. So my max-65 is wired to the SPI pins on EXT2, using PI9 on the can bus for CS. I tried to get this working with 2.1, but keep having low temp errors on bootup no matter what I've tried.

I've read over a related report here:
https://github.com/MarlinFirmware/Marlin/issues/23439

I have tried a few changes to config following that discussion, but nothing seems to change. Just FTY I used this info to get 2091 working fine:
https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/issues/179#issue-741317998

On a related side note, compiling errored out when trying to use the recommended config for hw spi temp... in configuration_adv.h, if I uncomment this it won't compile:
```
#define TEMP_SENSOR_FORCE_HW_SPI                // Ignore SCK/MOSI/MISO pins; use CS and the default SPI bus.
```

The error output mentions this should work, and it does compile without error like this:
```
#define TEMP_SENSOR_0_FORCE_HW_SPI                //trug Ignore SCK/MOSI/MISO pins; use CS and the default SPI bus.
```

Further, even with force hw spi defined, compiler warnings tell me TEMP_0_CS_PIN is being redefined to PH9 via the pins file, despite me forcing hwspi and `#define TEMP_0_CS_PIN  PI9` in configuration.h. So I had to comment out the line in my pins file regardless. Seems like the definition in configuration.h should take precedence, but IDK.

This is probably just a trivial macro error, but whatever broke hwspi2 for max-65 is beyond my ability to find and fix. Any help is appreciated.

### Expected behavior

Boot up and work fine, I suppose.

### Actual behavior

Min temp error halt right after bootup

### Version of Marlin Firmware

2.1 release

### Printer model

Anet a8

### Electronics

BTT GTR v1.0

### Add-ons

BTT TFT35 v3, adafruit max31865 breakout

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

Cura

### Host Software

SD Card (headless)

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[gtr1maxErrors.zip](https://github.com/MarlinFirmware/Marlin/files/9128369/gtr1maxErrors.zip)

**Comment by truglodite** [comment]
I recently tried this again on the latest release 2.1.1, and got the same min temp errors. It was a little surprising  configuring 2.1.1, since the latest GTR1.0 pins file has exp1&2 rotated 180* from the original definition for whatever reason.

It seems using hardware spi for the max breakout is somewhat obscure with weaker support vs soft spi. So my plan is to switch the max to soft spi using e2 pins (I have tmc2209). This approach seems to have worked out well for those that have done it. On the up side this will also allow me to switch to using the more reliable onboard micro sdcard slot (my old worn out LCD sdcard slot has been flakey lately).

**Comment by truglodite** [comment]
After switching to software spi as described above (E2 spi connectors w/ modified pins file), everything is working well using a 3-wire pt100.

I noticed that the LCD displays and menu scroll were much snappier than before. The LCD was always very laggy with hardware SPI, and often caused miss selections when scrolling through menus and it failed to update the display. The display lag also made other display items not work, like scrolling filenames during print, occasional glitch artifacts (random pixels etc). Now using software spi, all of the LCD features work perfect as far as I've observed.

I also observed, before when using hardware SPI, PID autotune kept erroring out. Originally I attributed that to upgrading from 40W to a 50W heater, but now with the display running faster I can see some over/undershoot that was not showing before. Now I wonder if this issue also caused timing issues with temp reports that may have confused the autotune algo. I've been busy IRL, but plan to give autotune another shot and see if that theory has any merits. I'll report back with my results to add more data to anyone who is stuck having to use hwspi for this.
